mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -200,22 +200,17 @@ abstract class ProbabilisticClassificationModel[
     if (!isDefined(thresholds)) {
       probability.argmax
     } else {
-      val thresholds: Array[Double] = getThresholds
-      val probabilities = probability.toArray
+      val thresholds = getThresholds
       var argMax = 0
       var max = Double.NegativeInfinity
       var i = 0
       val probabilitySize = probability.size
       while (i < probabilitySize) {
-        if (thresholds(i) == 0.0) {
-          max = Double.PositiveInfinity
+        // thresholds are all > 0, excepting that at most one may be 0
+        val scaled = probability(i) / thresholds(i)
+        if (scaled > max) {
```

Comment:
If `probability(i)` and `thresholds(i)` are both 0.0 here, we will have `scaled = NaN`. Maybe we can break out of the loop early if we encounter a zero threshold. BTW, this also begs the question of what the answer should be with an infinitely low probability and an infinitely low threshold - but I'm totally fine just predicting whatever threshold is zero in that case :D